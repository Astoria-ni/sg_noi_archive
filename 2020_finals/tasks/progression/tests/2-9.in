25 1000
-6 0 8 -1 -5 -5 2 5 -1 2 8 9 0 6 -1 -5 8 0 -3 -4 -5 0 -3 -4 -3
3 1 25
2 3 3 -5 0
1 13 20 -4 -3
1 6 19 10 -6
3 8 20
1 15 19 -8 4
1 18 24 7 8
3 8 25
1 4 6 -2 -6
2 3 23 9 2
3 19 23
2 3 3 -1 6
1 24 25 -2 10
3 15 15
3 5 12
3 8 8
3 20 21
3 8 8
1 6 8 2 6
2 20 22 -1 1
1 6 22 -4 0
1 7 14 1 5
2 3 10 5 10
3 24 25
2 14 23 10 6
2 1 25 -1 -10
2 7 24 -1 3
3 6 7
3 3 13
1 4 10 -8 -5
2 17 19 0 -4
1 21 24 9 8
1 12 23 8 8
3 24 25
1 14 17 -10 2
1 9 9 -4 -7
1 20 20 -8 -4
1 8 22 2 4
2 7 9 4 -1
2 9 20 -5 5
2 17 23 9 8
1 17 23 7 8
1 5 14 -7 5
1 16 20 1 -1
1 14 17 5 10
1 1 21 -6 4
1 8 12 -8 2
3 7 20
2 23 23 -1 7
1 22 22 0 -5
2 3 6 3 -10
2 19 25 -6 -5
2 10 19 5 -10
2 23 25 -2 3
2 22 22 6 -2
3 6 9
2 13 18 10 -4
1 16 23 3 5
2 8 20 2 -2
2 13 22 -3 -4
3 22 23
3 22 23
3 2 3
1 7 20 -6 -8
3 20 20
3 3 25
2 16 16 -7 8
3 9 18
3 25 25
3 16 22
3 20 22
3 7 13
1 1 13 5 -6
2 22 22 -8 0
1 4 21 -7 7
3 16 21
2 19 25 5 -2
2 6 7 6 4
1 18 19 -7 7
1 12 14 3 -5
2 12 15 0 -5
2 25 25 -6 -3
3 17 21
2 13 19 5 2
3 5 5
3 15 22
3 8 16
3 19 21
3 24 24
1 1 4 8 -6
1 12 18 1 3
1 11 16 -4 -8
1 22 24 -9 7
1 13 21 10 10
3 6 6
2 15 24 -3 2
2 16 23 7 -4
3 5 7
3 18 20
1 19 25 -4 -7
3 23 23
3 2 12
3 12 12
1 4 20 7 2
2 18 24 6 -5
2 11 20 6 4
3 6 21
3 13 24
1 7 19 -5 -5
3 21 24
1 19 21 2 5
2 3 7 -7 -3
2 13 13 0 9
1 6 23 -9 3
3 17 23
1 5 21 2 -3
3 12 15
3 12 16
3 19 23
2 5 15 -5 -8
2 17 17 2 2
1 25 25 6 -9
2 11 21 7 -7
1 17 17 -6 0
2 17 20 -10 8
2 12 15 7 8
1 7 25 -6 -7
2 19 20 -6 8
2 20 20 -2 6
2 8 17 -9 9
2 12 20 3 4
1 7 16 6 3
2 8 8 -7 9
2 6 7 -3 9
3 7 12
2 5 17 -7 -1
3 12 24
1 18 25 -7 4
2 8 22 -1 -6
3 18 19
2 15 20 -3 -4
1 12 20 -7 -2
2 22 23 -1 9
1 2 17 4 3
3 22 23
2 6 21 5 -8
2 23 25 1 4
2 12 21 1 -9
1 21 22 10 -9
1 1 2 10 10
1 5 10 2 -5
1 20 22 0 -1
2 3 20 -2 -3
2 3 18 0 -7
3 3 21
1 22 23 3 9
1 22 22 -1 7
1 23 24 -4 4
1 3 18 3 -10
1 9 20 -8 -3
3 16 22
1 11 20 -5 -9
1 1 18 8 2
1 24 25 5 7
3 20 20
1 12 17 4 10
1 23 23 1 -1
1 22 25 -2 1
1 6 18 -1 1
3 1 16
1 16 19 -6 7
1 5 14 1 9
1 5 5 1 -4
3 11 20
3 3 8
3 5 5
1 4 18 7 -8
3 16 18
1 14 24 9 5
1 13 16 -3 -1
2 15 19 -6 0
1 25 25 -4 8
1 4 11 5 -10
3 5 6
3 24 24
2 20 21 10 6
1 8 10 -5 9
3 3 24
1 2 23 -7 3
2 16 16 10 3
1 24 24 -6 -2
3 9 22
3 12 15
2 11 14 5 4
3 19 19
2 20 25 9 6
1 11 11 7 -10
1 1 4 -8 -5
1 15 18 3 -2
1 24 25 -5 -3
3 23 24
3 21 24
2 6 10 6 3
1 25 25 10 1
3 4 9
3 23 23
1 10 21 -7 -5
1 11 20 -2 0
3 19 21
3 12 17
1 24 25 5 -2
2 22 24 -4 6
3 19 24
3 2 25
3 23 24
3 6 7
1 11 17 10 -3
1 3 14 -10 -6
2 8 9 5 -9
3 15 16
2 18 18 -6 6
2 16 19 -7 -2
2 12 19 3 -5
3 4 15
2 13 21 -8 6
1 7 20 -1 8
2 9 24 8 -2
1 24 25 -5 8
1 13 18 8 -5
3 12 22
2 10 25 3 1
1 2 17 0 -6
2 9 10 0 -10
3 2 8
1 18 23 -2 0
3 9 21
1 8 9 10 -2
2 7 23 -10 1
3 13 18
1 23 23 3 9
2 23 25 -6 -7
3 24 24
3 15 17
1 9 9 3 -4
2 19 25 -7 -2
3 12 12
2 18 22 9 0
2 3 16 -6 -4
1 16 19 10 2
3 22 23
2 19 21 -8 -2
3 9 17
1 14 14 1 -8
2 15 16 8 -8
3 4 21
2 10 18 5 4
3 7 18
3 23 23
1 14 19 8 5
2 9 9 -9 -3
3 7 13
1 13 22 -6 2
3 9 12
2 25 25 6 -9
1 22 23 10 2
3 19 22
3 20 21
1 21 21 1 3
2 25 25 4 2
3 25 25
3 2 24
2 11 14 -1 9
1 14 19 0 -5
2 9 20 -7 -8
2 20 21 0 -2
2 1 7 -9 -4
1 13 17 0 9
3 7 8
3 14 14
2 4 4 0 8
2 13 23 4 0
2 14 22 9 6
3 16 19
2 2 2 -7 4
3 18 21
2 22 25 -1 -5
1 13 13 -10 -5
2 3 6 9 -8
2 11 11 8 1
1 8 10 -2 6
3 17 18
3 7 9
1 18 21 6 5
1 17 21 -3 -4
2 22 22 1 -10
3 23 24
1 23 23 7 -2
2 10 20 -4 2
1 12 13 -8 -4
2 1 21 5 1
2 17 23 9 5
3 14 17
2 24 24 2 -5
2 3 21 5 1
2 5 25 -3 -4
1 25 25 4 2
1 11 24 -1 -8
1 3 15 0 -5
1 22 24 5 -6
1 8 12 -3 1
1 17 22 6 8
3 1 16
1 19 22 -8 -4
3 6 18
2 11 25 -3 -2
1 1 13 6 8
2 19 25 8 -1
1 9 17 -7 -8
2 12 24 8 -3
2 21 23 2 3
2 25 25 7 -8
3 22 22
1 22 23 6 6
1 17 18 -9 -5
2 23 23 -10 9
3 18 24
3 21 24
3 18 18
2 2 12 7 7
1 14 25 8 -10
3 25 25
2 18 18 -2 -10
1 22 25 -3 0
2 10 23 -1 -4
2 10 20 6 -2
1 9 21 -4 4
1 8 20 4 8
1 13 15 -4 5
1 9 21 2 -5
1 13 15 1 6
2 22 22 -6 8
3 19 20
1 24 25 8 9
3 18 21
3 8 13
2 4 12 -8 -5
3 3 17
3 10 16
3 15 24
1 13 25 8 -7
2 25 25 8 -9
2 21 25 4 -3
1 17 20 5 -1
3 8 9
2 11 25 8 2
3 16 16
1 19 21 -9 -3
2 5 24 -7 -8
1 18 18 6 9
1 4 24 3 6
1 16 17 7 -6
3 19 23
1 14 23 -4 -1
1 6 17 -4 3
1 4 8 3 10
3 12 13
3 15 20
2 23 25 7 -2
2 10 19 9 4
2 21 21 -8 8
2 15 15 -7 6
3 7 15
1 1 7 4 9
3 5 5
2 16 20 -4 10
1 7 7 -7 -6
3 12 25
3 7 18
1 4 13 -2 -5
2 4 23 -10 -8
3 8 18
3 1 24
3 8 10
1 6 18 -6 3
3 5 6
1 7 18 0 -1
3 25 25
3 12 22
1 9 23 2 -8
1 13 14 1 -3
3 6 18
2 17 18 10 0
3 18 19
3 1 19
1 17 21 -1 1
2 10 24 -2 4
2 8 19 -9 -5
1 12 22 9 -4
3 23 24
2 9 18 -1 -1
1 14 14 -5 6
3 8 11
1 12 19 2 -6
2 18 18 3 6
1 5 24 8 2
2 5 9 8 -1
2 11 11 1 -3
2 10 22 -10 -1
2 2 4 5 2
2 24 25 9 -2
1 12 15 1 7
1 15 17 8 9
2 13 16 1 2
2 24 24 3 10
1 12 15 10 3
3 18 19
2 23 25 -4 -3
3 20 25
2 16 16 -3 -8
1 7 23 8 -9
3 5 8
3 19 19
2 24 25 -7 -7
2 24 24 -7 -8
2 4 17 -7 -9
3 16 21
2 12 20 1 3
1 6 21 3 10
1 18 23 8 10
2 9 17 4 -7
1 17 21 7 8
1 3 23 -5 4
3 18 22
3 5 9
3 17 22
2 11 12 4 -6
1 15 23 -1 2
1 9 22 -8 6
3 11 15
1 7 12 5 2
1 5 20 -9 -5
2 24 24 -7 2
3 4 19
3 9 12
2 18 24 7 2
3 1 4
2 25 25 6 -8
1 6 22 -2 -5
2 12 20 -6 0
3 6 17
3 11 23
2 9 14 -8 -2
1 13 18 2 1
3 8 18
2 21 21 6 1
3 20 22
3 25 25
2 12 24 -7 -5
3 14 18
1 17 20 -5 1
3 17 23
1 5 21 6 -9
1 9 15 6 -3
2 12 22 3 5
3 25 25
2 14 19 3 -8
2 4 19 -3 7
2 16 21 -8 8
3 9 14
2 14 21 2 -8
2 1 11 -5 -10
1 5 14 2 5
3 11 13
2 14 19 -7 8
2 8 11 -1 -2
3 5 15
2 12 19 -10 -5
2 4 18 8 1
1 15 23 -5 -7
1 2 25 -8 -2
3 6 8
2 3 10 -5 9
1 14 19 3 -3
2 23 23 8 10
1 19 23 7 1
3 7 21
1 7 10 3 -7
2 17 20 -9 3
3 4 4
3 13 19
1 13 25 -6 6
3 2 24
2 11 13 9 5
1 21 24 7 -8
1 17 22 4 -1
3 16 21
3 18 22
3 14 24
2 13 22 -5 -2
2 11 21 -2 -9
3 20 20
1 18 22 -5 -9
3 14 15
1 22 24 3 -9
3 22 22
2 21 25 4 0
2 22 25 9 -4
1 14 23 -5 0
3 4 24
3 4 17
3 22 24
3 15 18
1 7 21 1 3
2 23 25 3 0
3 12 24
1 8 8 2 5
3 24 25
1 18 18 -7 -3
1 21 24 0 10
3 16 18
1 2 8 0 -10
1 25 25 -7 4
3 6 16
3 23 24
2 12 14 -5 -7
1 13 24 6 9
3 16 19
2 24 25 -3 -5
3 8 17
2 9 10 0 -3
2 5 14 8 -3
2 13 23 -9 -9
2 1 10 -7 1
3 21 21
1 6 14 8 -10
2 1 17 10 1
1 2 5 9 -9
2 14 22 8 2
2 13 14 -4 9
2 16 18 5 -4
1 4 20 -1 -10
1 7 14 4 1
1 1 18 -7 -10
1 10 12 -2 -6
2 16 25 3 3
1 19 21 5 -5
3 6 21
2 20 21 10 -10
2 14 14 5 3
1 7 12 -8 2
1 11 15 -6 0
3 11 13
1 17 20 -2 2
3 10 24
1 23 25 3 -4
3 12 14
3 7 20
3 2 5
1 24 25 -2 -6
3 8 16
3 3 10
2 24 25 -9 -6
3 13 25
2 10 21 -3 -6
2 7 7 -5 -8
2 8 23 -8 10
1 12 12 -7 -6
1 16 22 -4 -1
2 5 11 -8 7
3 1 24
1 21 22 -6 8
3 18 22
2 24 24 -1 -10
2 12 13 1 -3
3 23 25
2 18 22 -9 8
3 15 22
1 13 24 7 -4
1 16 20 9 7
2 23 24 4 -2
2 17 20 -4 -9
2 14 24 8 4
1 1 25 8 8
2 24 24 -4 9
1 7 8 -1 5
1 2 11 -8 10
3 6 21
2 24 25 -7 -7
3 8 14
2 11 21 -9 1
3 9 15
1 12 12 8 6
1 1 18 7 0
1 3 22 0 -10
3 20 20
1 17 25 0 -2
3 1 22
3 17 19
2 18 24 -6 5
3 7 14
3 25 25
3 23 25
3 6 8
2 4 21 -6 -9
1 7 23 5 -4
3 5 12
3 18 23
3 17 21
2 16 25 4 10
1 14 14 2 -3
2 1 9 -1 7
1 15 20 -8 -3
2 18 23 -10 -5
3 7 23
2 5 18 3 1
2 8 21 -5 -9
1 18 23 0 8
2 7 22 -6 3
2 16 19 -1 -10
2 11 22 -5 1
2 14 20 -1 -10
2 24 25 0 9
2 3 6 -5 6
2 4 16 -3 -9
3 8 18
3 12 17
3 23 23
2 9 25 -5 7
2 11 25 8 -10
3 20 23
3 9 19
2 21 25 -6 3
2 9 23 4 -5
3 20 23
3 20 24
2 9 18 1 0
2 3 8 -6 3
1 1 14 5 0
1 2 19 -8 -5
2 10 22 7 -2
1 17 17 -5 -1
1 4 19 6 -6
1 2 25 -8 -7
3 4 7
3 21 25
1 24 24 -8 5
3 21 21
3 4 15
2 20 24 1 2
3 14 14
1 16 18 -4 -6
1 5 15 6 -4
3 7 21
2 12 16 -3 1
1 23 23 -7 4
1 20 24 -1 -7
1 14 15 1 7
2 6 11 3 -5
3 13 21
1 23 23 3 -10
3 16 20
3 4 14
3 1 3
3 2 22
3 21 25
3 2 19
1 25 25 6 -2
2 1 6 -9 -3
3 25 25
2 22 24 5 -10
3 19 24
3 3 13
3 25 25
1 13 17 -10 -5
2 2 19 -3 8
2 14 22 -2 3
3 3 17
1 13 25 -2 1
3 3 5
1 13 15 -3 -3
1 20 22 -3 -9
3 16 21
3 14 21
3 23 24
3 15 16
1 1 15 3 3
2 25 25 2 -4
2 4 16 -7 -2
3 23 23
1 19 20 -6 4
1 20 20 -6 0
1 8 19 -7 0
2 8 8 -7 4
3 5 20
1 25 25 10 2
3 15 15
1 4 11 3 1
1 4 7 1 -9
2 4 9 7 -10
1 8 24 9 10
2 25 25 3 1
2 9 23 5 3
3 24 24
3 18 18
1 23 23 -3 -5
2 1 23 2 -8
2 16 19 -4 -6
1 3 24 5 0
1 14 17 2 3
1 7 15 1 -9
1 12 24 2 4
3 21 22
1 10 11 -1 -3
3 10 14
2 14 23 5 -9
1 5 5 -10 -6
3 7 15
2 15 20 7 -4
3 8 19
3 24 24
2 5 11 7 7
1 22 24 3 -9
2 16 22 3 -2
1 8 18 1 -5
2 17 24 8 5
2 12 23 3 -6
2 1 9 -10 -9
3 21 24
3 8 15
3 3 3
2 17 21 -2 5
1 13 24 5 -10
1 21 23 10 -7
2 11 20 3 -9
3 6 6
1 5 8 -4 5
2 1 21 -1 -4
2 2 10 -3 -9
3 12 12
1 13 13 1 2
3 25 25
1 24 24 4 -2
2 13 16 -7 6
1 14 16 -3 -7
1 4 16 -1 8
2 9 24 9 6
3 4 11
1 19 20 -5 4
2 23 23 -10 -10
3 6 16
2 18 24 9 2
2 21 24 -9 -10
3 21 22
2 1 21 -6 -8
2 25 25 9 2
3 14 15
2 8 23 -9 -8
1 20 22 -6 -10
2 5 5 2 -7
3 7 25
3 9 14
3 6 12
2 24 25 -3 -10
1 22 23 -5 -7
1 23 23 4 9
3 1 7
3 22 23
1 22 24 10 -8
1 7 21 -9 -8
2 2 13 -7 -1
3 9 11
3 24 24
2 14 16 4 -2
1 5 6 -10 -2
2 6 19 0 -4
1 8 20 6 -3
2 17 24 6 10
3 23 23
3 23 23
3 12 15
3 4 23
2 3 7 -8 7
1 23 23 5 10
3 25 25
1 1 25 2 -8
1 12 19 -10 1
3 23 25
1 9 19 -4 4
2 2 2 10 -9
2 20 24 2 -10
1 13 20 7 6
1 4 7 -1 -3
3 23 23
3 1 9
2 9 16 10 3
2 20 21 3 10
2 11 18 0 -2
1 15 24 5 -1
1 2 19 6 -8
2 20 25 -4 9
2 18 21 3 6
1 16 19 9 2
3 18 19
2 7 11 -5 -8
1 25 25 -6 3
3 23 24
2 6 20 10 -4
1 19 22 -7 -8
3 4 11
3 8 23
3 9 20
3 25 25
1 22 25 1 7
3 13 15
2 6 16 -9 9
2 18 22 -9 -2
3 5 13
2 2 9 8 0
2 21 24 5 -10
3 11 11
2 11 14 -6 1
3 5 7
3 5 15
2 24 24 3 0
3 14 19
2 15 23 9 -7
1 12 22 1 -3
2 19 22 1 -8
1 24 24 2 9
2 3 12 1 -2
3 4 5
3 9 12
1 20 21 -2 -10
1 3 9 -6 4
1 6 13 8 -4
2 7 7 7 9
2 8 20 -8 5
1 25 25 4 5
2 13 14 -6 1
1 13 24 4 7
2 18 24 10 -4
2 6 6 3 -4
3 2 12
1 15 19 0 5
1 6 24 -7 -1
1 4 20 3 3
2 22 25 10 10
2 14 17 6 -8
1 2 25 0 6
3 4 5
3 15 23
3 22 25
3 14 19
1 23 25 7 7
2 20 24 -6 -7
2 19 21 10 7
3 20 22
3 25 25
3 16 16
1 22 25 6 10
2 8 19 -8 4
3 10 12
3 18 25
2 6 10 5 0
1 7 22 2 1
3 11 14
2 13 23 4 10
3 17 19
2 24 25 -7 8
2 23 24 5 -4
2 19 19 3 9
3 15 16
2 12 17 -2 9
1 10 20 4 9
1 6 16 8 4
1 8 13 0 7
3 10 23
3 19 21
1 19 21 2 -4
3 11 21
1 22 24 -10 -9
3 4 22
1 23 25 -1 -10
2 21 23 8 6
3 14 18
1 20 23 -10 0
1 9 13 -5 -6
1 24 24 7 -6
2 5 13 -7 4
3 3 25
1 12 18 9 -8
2 7 22 -5 8
1 17 21 -3 8
1 12 20 -8 -6
1 15 19 5 -6
2 4 23 -2 1
3 21 25
1 2 3 4 -4
1 7 15 7 -2
1 20 25 2 6
3 11 23
3 4 13
2 15 22 -9 9
1 19 21 -5 -8
1 7 17 -3 -8
2 9 15 2 8
3 25 25
3 12 23
2 11 24 -6 -1
2 8 13 -1 -4
3 2 2
3 12 19
1 19 25 3 -1
2 22 25 -1 10
1 20 25 0 4
2 4 25 5 -8
1 8 23 -7 10
3 6 22
3 20 23
3 19 21
1 7 9 8 9
3 17 20
1 2 25 -8 6
3 4 10
2 7 25 -5 6
2 14 24 -10 1
1 7 22 10 4
1 25 25 -4 -5
1 24 24 3 -9
2 17 24 1 -8
3 1 12
3 11 11
3 6 16
3 21 24
2 25 25 -5 4
3 19 20
3 19 21
2 6 14 8 2
1 25 25 2 -5
3 24 24
1 19 19 7 -8
3 20 20
2 11 11 -6 5
1 5 6 -7 4
2 18 18 1 6
3 10 12
1 21 25 0 6
3 6 16
1 20 21 -3 -2
3 22 22
2 20 23 8 -10
1 21 23 -10 7
2 5 13 10 7
2 10 17 -8 3
1 18 25 6 7
3 15 15
2 15 15 4 -10
2 9 11 -7 0
1 22 23 -2 5
2 10 13 -6 3
2 22 22 4 9
3 20 25
2 1 21 -8 7
3 4 16
3 11 11
3 21 22
1 25 25 8 3
1 6 23 4 -3
1 4 24 -2 10
2 8 11 -10 -1
3 9 11
3 24 25
3 11 17
3 2 12
3 7 19
3 2 15
1 21 24 9 -2
2 9 11 -9 3
1 2 12 9 10
2 16 18 -2 8
3 7 20
2 16 18 -8 -9
2 18 21 4 -10
3 9 16
1 16 21 9 4
2 14 24 -3 10
1 8 23 -9 0
3 12 25
3 17 25
1 18 21 10 6
3 23 24
3 24 25
2 9 17 2 0
1 14 23 -7 9
3 23 24
1 13 18 -10 -5
2 1 16 -1 -4
2 15 25 5 -6
2 23 24 2 5
3 1 25
