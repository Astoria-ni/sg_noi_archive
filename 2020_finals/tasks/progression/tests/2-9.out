3
2
2
5
1
8
1
2
1
2
2
7
2
4
2
2
2
2
1
8
4
1
4
2
5
4
3
1
5
4
2
1
1
2
3
1
3
1
10
8
4
2
2
3
2
3
6
2
2
16
3
1
5
3
3
1
2
2
1
5
3
2
1
2
3
4
1
2
3
3
4
2
2
2
4
6
3
7
6
1
3
1
2
7
10
9
1
4
3
2
2
1
6
2
1
4
4
2
2
2
3
3
3
1
2
2
1
1
2
3
3
9
3
5
2
1
5
2
2
5
1
5
5
11
20
3
2
1
7
4
2
4
2
3
2
3
2
1
4
3
3
3
2
2
2
2
6
9
4
2
1
5
4
1
6
2
4
3
4
1
3
4
4
3
4
1
2
1
4
4
2
3
4
2
2
3
2
2
3
1
3
2
3
2
3
2
2
3
2
7
3
2
5
8
3
5
1
5
2
2
1
2
2
6
4
5
6
9
5
1
4
9
4
4
4
2
1
6
1
6
2
2
6
2
6
2
6
1
3
5
1
11
3
4
4
2
2
1
8
1
1
1
2
2
3
5
1
3
4
1
1
1
1
5
8
2
2
12
6
5
4
2
3
1
1
1
4
9
1
2
1
4
2
2
6
11
10
1
3
8
1
3
5
2
2
4
5
2
3
4
4
2
1
1
3
3
4
3
2
3
2
3
3
3
9
3
5
7
1
4
1
6
15
4
3
4
3
6
1
7
4
2
3
1
1
2
4
1
1
2
13
1
2
3
2
6
4
8
4
3
3
10
7
2
2
2
14
